src/UI/Component/Item/AppointmentItem.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Item;
+
+/**
+ * Interface AppointmentItem
+ * @package ILIAS\UI\Component\Panel\Listing
+ */
+interface AppointmentItem extends Item {
+	/**
+	 * Get the starting point of the appointment.
+	 * @return \ilDateTime
+	 */
+	public function getFrom();
+
+	/**
+	 * Get the ending point of the appointment.
+	 * @return \ilDateTime
+	 */
+	public function getEnd();
+
+	/**
+	 * Get the color of the calendar containing the item as color code (hex).
+	 * @return string
```

Comment:
> I think we need both. On the calendar workshop we talked about the possibility to create custom calendar categories, as far as I remember. This will be the moment we need the possibility to just set a color at the latest.
I disagree. The number of colors that are distinguishable by users is not very high. Timon/Yvonne are currently looking into this, but I am quite sure it will not be more than 32. Any additional color does not make any sense. If you have a larger number of categories you will be forced to re-use colors imo. And I do not see a big issue here, since we are not allowed to "rely on color alone" when presenting classification information due to accessibility reasons anyway.

I very much agree that we need a general color palette concept. Besides calendar and tags the upcoming letter avtars are another case for this: http://www.ilias.de/docu/goto_docu_wiki_wpage_2459_1357.html